src/sentry/tasks/store.py

```diff
@@ -417,16 +425,104 @@ def save_attachment(event, attachment):
+
+def crashreports_exceeded(current_count, max_count):
+    if max_count == STORE_CRASH_REPORTS_ALL:
+        return False
+    return current_count >= max_count
+
+
+def get_stored_crashreports(cache_key, event, max_crashreports):
+    # There are two common cases: Storing crash reports is disabled, or is
+    # unbounded. In both cases, there is no need in caching values or querying
+    # the database.
+    if max_crashreports in (0, STORE_CRASH_REPORTS_ALL):
+        return max_crashreports
+
+    cached_reports = default_cache.get(cache_key, None)
+    if cached_reports >= max_crashreports:
+        return cached_reports
+
+    # Fall-through if max_crashreports was bumped to get a more accurate number.
+    return EventAttachment.objects.filter(
+        group_id=event.group_id, file__type__in=CRASH_REPORT_TYPES
+    ).count()
+
+
+def save_attachments(cache_key, event):
+    """
+    Persists cached event attachments into the file store.
+
+    This method checks whether event attachments are available and sends them to
+    the blob store. There is special handling for crash reports which may
+    contain unstripped PII. If the project or organization is configured to
+    limit the amount of crash reports per group, the number of stored crashes is
+    limited.
+
+    :param cache_key: The cache key at which the event payload is stored in the
+                      cache. This is used to retrieve attachments.
+    :param event:     The event model instance.
+    """
+    if not features.has("organizations:event-attachments", event.project.organization, actor=None):
+        return
+
+    attachments = attachment_cache.get(cache_key) or []
+    if not attachments:
+        return
+
+    # The setting is both an organization and project setting. The project
+    # setting strictly overrides the organization setting, unless set to the
+    # default.
+    max_crashreports = get_max_crashreports(event.project)
+    if not max_crashreports:
+        max_crashreports = get_max_crashreports(event.project.organization)
+
+    # The number of crash reports is cached per group
+    crashreports_key = get_crashreport_key(event.group_id)
+
+    # Only fetch the number of stored crash reports if there is a crash report
+    # in the list of attachments. Otherwise, we won't require this number.
+    if any(attachment.type in CRASH_REPORT_TYPES for attachment in attachments):
+        cached_reports = get_stored_crashreports(crashreports_key, event, max_crashreports)
+    else:
+        cached_reports = 0
+    stored_reports = cached_reports
+
+    for attachment in attachments:
+        # If the attachment is a crash report (e.g. minidump), we need to honor
+        # the store_crash_reports setting. Otherwise, we assume that the client
+        # has already verified PII and just store the attachment.
+        if attachment.type in CRASH_REPORT_TYPES:
+            if crashreports_exceeded(stored_reports, max_crashreports):
+                continue
```

Comment:
can this be a `break`?

Reply:
We still need to continue with the next attachments, since there might be non-crash-report attachments in the list.